### PR TITLE
resources-report: allow whitelisting, add build resources to whitelist

### LINKED ISSUE #11005: resources-report: add whitelist functionality for certain resources
Follow-up to https://github.com/sourcegraph/sourcegraph/pull/10784 - the first iteration of the action reports everything it finds that was created in the last x hours. However, some autoscaling services can generate a lot of noise in the report - most notably buildkite, where fluctuations in resources used is expected.

This filtering could be done via some custom labelling scheme (for example, `"resources-report": "ignore"`) or similar

### PR DESCRIPTION
**Pull request by bobheadxi**
closes #11005

* add `--gcp.whitelist` and `--aws.whitelist` to indicate resources to ignore (based on labels and tags, respectively)
* update output to reflect this (summary in slack message, drop whitelisted resources from report)

sample output: https://sourcegraph.slack.com/archives/C013V78N0Q2/p1590769797000400

### CODE REVIEW

File: internal/cmd/resources-report/main.go [hunk 1]
```diff
@@ -101,3 +114,12 @@ func reportString(resources Resources) string {
 	}
 	return output
 }
+
+func stringToMap(str string) map[string]string {
```
Comment (slimsag) [inline internal/cmd/resources-report/main.go:118]: Rename to `csvToMap` maybe? And add docstring indicating the format it takes

File: internal/cmd/resources-report/resource.go [hunk 2]
```diff
@@ -34,6 +36,18 @@ func (r Resources) Swap(i, j int) {
 	r[j] = tmp
 }
 
+// Unwhitelisted only unwhitelisted resources
```
Comment (slimsag) [inline internal/cmd/resources-report/resource.go:39]:
```suggestion
// Unwhitelisted returns only unwhitelisted resources
```

File: internal/cmd/resources-report/main.go (hunk 1 above)
Comment (bobheadxi) [reply internal/cmd/resources-report/main.go:118]: 🤔 is `key:value,key:value` a csv format?

other than that this definitely needs a docstring, will add!

Comment (slimsag) [reply internal/cmd/resources-report/main.go:118]: It is :) only some CSV parsers support colons, though.

Comment (slimsag) [reply internal/cmd/resources-report/main.go:118]: (as long as you use something other than the word `string` to describe it, I'm happy. My first impression was "How do you turn a string into a map?! Like one character to character? Or `"thestring": "thestring"`?

Comment (bobheadxi) [reply internal/cmd/resources-report/main.go]: Hopefully 0129fdd41789d97fd81d422e0cfac569bc1ac326 clears it up a bit!

**Pull request merged**